src/jit/compiler.cpp

```diff
@@ -5948,7 +5948,12 @@ int Compiler::compCompileHelper(CORINFO_MODULE_HANDLE            classPtr,
         goto _Next;
     }
 
+#ifdef FEATURE_CORECLR
     if (fgHasBackwardJump && (info.compFlags & CORINFO_FLG_DISABLE_TIER0_FOR_LOOPS) != 0 && fgCanSwitchToTier1())
```

Comment:
The condition is a bit long, we can merge it with `!FEATURE_CORECLR` or use the same order for them:
`if (fgHasBackwardJump && fgCanSwitchToTier1() && ((info.compFlags & CORINFO_FLG_DISABLE_TIER0_FOR_LOOPS) != 0))`